hikari/internal/enums.pyi

```diff
@@ -33,11 +33,19 @@
 __all__ = ["Enum", "Flag"]
 
 import enum as __enum
+from typing import Any as __Any
 from typing import Iterator as __Iterator
 from typing import Sequence as __Sequence
 from typing import TypeVar as __TypeVar
 from typing import Union as __Union
 
+class deprecated:
+    value: __Any
```

Comment:
As much as I would love to type it as a function (i did try), mypy will treat it as the correct type, and this will cause issues in `enums.py` when we try to access values on it or try to `isinstance` it.

Reply:
why does that matter when `deprecated` isn't meant to be seen nor handled outside of enum.py

Reply:
That is true. Forgot to comment, but I tested using `reveal_type` and the returned type seems to be correct even without these changes

Reply:
It isn't

```py
from hikari.internal import enums
import warnings

warnings.simplefilter("ignore", category=DeprecationWarning)

class FooEnum(int, enums.Enum):
    FOO = 1

    BAR = enums.deprecated(FOO, removal_version="3.1.0")


print("type:", FooEnum.FOO, type(FooEnum.FOO))
print("member:", FooEnum.FOO.value, type(FooEnum.FOO.value))
print("member:", FooEnum.BAR, type(FooEnum.BAR))
print("value:", FooEnum.BAR.value, type(FooEnum.BAR.value))

import typing
if typing.TYPE_CHECKING:
    reveal_type(FooEnum.FOO.value)
    reveal_type(FooEnum.BAR.value)

```
```py
hikari\internal\time.py:93:1: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
test.py:19:17: note: Revealed type is "Literal[1]?"
test.py:20:17: note: Revealed type is "hikari.internal.enums.deprecated"
Found 1 error in 1 file (checked 1 source file)
```

Reply:
Oh, thats interesting. What python version are you using? Because that same code works for me:

```
nox > Running session mypy
nox > Re-using existing virtual environment at .nox/mypy.
nox > python -m pip install -r requirements.txt -r speedup-requirements.txt -r server-requirements.txt -r dev-requirements/mypy.txt -r dev-requirements/formatting.txt
nox > stubgen hikari/__init__.py hikari/api/__init__.py hikari/events/__init__.py hikari/impl/__init__.py hikari/interactions/__init__.py -o . --include-private --no-import
Processed 5 modules
Generated files under hikari/
nox > mypy -p hikari --config pyproject.toml
hikari/intents.py:420:13: note: Revealed type is "Literal[hikari.intents.Intents.DEPRECATED]?"
Success: no issues found in 99 source files
nox > mypy -p examples --config pyproject.toml
hikari/__init__.pyi:76: note: In module imported here,
examples/hello_world.py:14: note: ... from here:
hikari/intents.py:420:13: note: Revealed type is "Literal[hikari.intents.Intents.DEPRECATED]?"
Success: no issues found in 6 source files
nox > Session mypy was successful.
```

*Had to just add a random deprecated enum member to intents because mypy refused to pick up the stubs for hikari.internal after installing it*

Reply:
Fixed :)